torch/csrc/jit/codegen/cuda/runtime/random_numbers.cu

```diff
@@ -110,3 +109,63 @@ __device__ double uniform(unsigned int x, unsigned int y) {
       (unsigned long long)x ^ ((unsigned long long)y << (53 - 32));
   return z * kRan2Pow53Inv + (kRan2Pow53Inv / 2.0);
 }
+
+
+namespace at {
```

Comment:
does this have to be in that at namespace? this could lead to subtle odr violations which would likely be hard to troubleshoot

Reply:
I guess not, but I don't understand why it's a problem.  Classes are always defined in headers without odr violation as long as member functions in the header are inline.  Why is it a problem in this case?

Reply:
Is it because this stuff will become a distinct .so from the similarly-defined stuff in ATen?

Reply:
the headers generally work since the definitions are identical. If the definitions diverge, even in seemingly harmless ways then it's undefined what happens - in practice the issues are hard to troubleshoot since there's no toolchain support today.

PS. even with headers you can get ODR issues if the translation units are compiled with different flags or with different macros

Reply:
> Is it because this stuff will become a distinct .so from the similarly-defined stuff in ATen?

No, that's an ABI compatibility thing. The potential problem is if anyone ends up including both the aten header and the nvfuser_resources - it seems a bit stretched in the current setup, but let's say someone wants to prototype some CUDA changes by hand. Or if we end up adding support for real header includes to hvrtc.

Reply:
If CPU-compiled and nvrtc-compiled instances are ABI-incompatible, aren't we screwed regardless because the CPU instances will be bitcopied into the kernel instances?  How does putting the definition to be compiled by nvrtc into a different namespace save us from that?

Reply:
it's the other way around: putting things in the same namespace may lead to extra issues even if we're layout/abi compatible.